catalog/resource_grant.go

```diff
@@ -0,0 +1,179 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/catalog/permissions"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// diffPermissionsForPrincipal returns UnityCatalogPermissionsDiff of this permissions list with `diff` privileges removed
+func diffPermissionsForPrincipal(principal string, pl catalog.PermissionsList, existing catalog.PermissionsList) (diff []catalog.PermissionsChange) {
+	// diffs change sets for principal
+	configured := map[string]*schema.Set{}
```

Comment:
Maybe I'm misreading this, but because this function only ever operates on one principal, this map will only have one key defined? If so, we can just use `*schema.Set` as opposed to a map. This applies to the other data structures throughout this function.

Reply:
Ping here